modules/signals/src/signal-method.ts

```diff
@@ -30,10 +28,13 @@ export function signalMethod<Input>(
   const sourceInjector = config?.injector ?? inject(Injector);
 
   const signalMethodFn = (
-    input: Input | Signal<Input>,
+    input: Input | (() => Input),
     config?: { injector?: Injector }
   ): EffectRef => {
-    if (isSignal(input)) {
+    if (typeof input !== 'function') {
```

Comment:
Switched the if condition to eliminate the possibility of a function in the else condition.

With `Input | (() => Input)` and a `typeof input === 'function', TypeScript would not narrow done the type union.